### PR TITLE
ENH: Add annotations to `np.core.multiarray` part 4/4

### PR DESCRIPTION
**Pull request by BvB93**
Follow up on https://github.com/numpy/numpy/pull/19237

This PR adds annotations for the final two objects in the `np.core.multiarray` namespace: 
* The `nested_iters` function 
* The `nditer` class.

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -504,6 +504,8 @@
 
 add_newdoc('numpy.core', 'nested_iters',
     """
+    nested_iters(op, axes, flags=None, op_flags=None, op_dtypes=None, order="K", casting="safe", buffersize=0)
```
Comment (BvB93) [inline numpy/core/_add_newdocs.py:507]: I don't think there is any way here around the "line too long" complaints, is there?

Comment (charris) [reply numpy/core/_add_newdocs.py:507]: Hmm. Don't know of one for sure, but it looks like line continuation works inside the quotes
```
>>> """
... Hello \
... World
... """
'\nHello World\n'
```
The problem might be the indentation of the next line cannot be aligned with the opening "(", but that might not matter.

Comment (BvB93) [reply numpy/core/_add_newdocs.py]: Good idea, it seems adding the line-break did the trick:
https://22298-908607-gh.circle-artifacts.com/0/doc/build/html/reference/generated/numpy.nested_iters.html?highlight=nested_iters#numpy.nested_iters

**Comment by charris** [comment]
Thanks Bas.

**Pull request merged**